src/data_structures.rs

```diff
@@ -226,11 +237,28 @@ impl<'a, F: Field> LabeledPolynomial<F> {
     }
 }
 
+impl<F: Field> LinearCombination<F> for Polynomial<F> {
+    type Item = Polynomial<F>;
+
+    fn from_lc(items: Vec<Self::Item>, base: F, step: usize) -> Self::Item {
```

Comment:
I'd opt for a more general implementation here: 
We introduce the formal LCs for polynomials (as was there before, I believe), and adapt from_lc accordingly. For that we might need an associated struct to the LC trait, which carries the formal linear combinations.

Reply:
The reason for a more general implementation is simple: Depending on the situation, the coefficients of a linear combination are "constructed" differently. Sometimes they are just powers of a random challenge, as is the case when building random linear combination or the query-point dependent ones. However, a general treatment seems to be more natural to me, also having a future extensions of our proof functions to LCs in mind.
If we keep with this interface (with base and step), I'd rather call the function `build_lc()`.